src/JsonDeserializationVisitor.php

```diff
@@ -120,6 +120,13 @@ public function visitProperty(PropertyMetadata $metadata, $data)
             throw new RuntimeException(sprintf('Invalid data %s (%s), expected "%s".', json_encode($data), $metadata->type['name'], $metadata->class));
         }
 
+        if ($metadata->inline){
+            if (!$metadata->type) {
```

Comment:
can you use here `!==`?

Reply:
Let's keep the same logic that is few lines below. I'm not perfectly sure what value I can expect here.